### PR TITLE
Add files via upload

### PR DESCRIPTION
**Pull request by andr3wsw4g**
Made a number of changes to improve readability

### CODE REVIEW

File: fourierSeries.m [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    if p ~= length(xt);
+        break
+    end
+    end
+end
+end
+
+function coefficients = fourierapp(xt,T)
+if T <= 1
+
```
Comment (MickeyDang) [inline fourierSeries.m:43]: why so many spaces?

File: fourierSeries.m [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+for k = 1:(T)
+    for j = 1:(T)
+        coefficients(k) = coefficients(k) + 1/T*xt(j)*(exp(-1.0i*j*wo*(k)));
+    end
+end
+end
+end
+
+
+
```
Comment (MickeyDang) [inline fourierSeries.m:57]: Why so much white space?

File: invFourier.m [hunk 3]
```diff
@@ -0,0 +1,9 @@
+function OFunction = invFourier(coefficients, time)
```
Comment (MickeyDang) [inline invFourier.m:1]: what is `0Function` ?

File: testconvolute.m [hunk 4]
```diff
@@ -1,6 +1,43 @@
...
-a = length(xt);
-Z = convolute(xt,ht)
-Z(1,1) = Z(1,1)
-a = conv(0.01*xt,ht)
\ No newline at end of file
+xt = ones(1,100);%function x(t)
+ht = 0.01:0.01:2;%function h(t)
+Zt = convolute(xt,ht);%function Z(t), convoluted function
+cCheck = conv(0.01*xt,ht);%Our check for our test
+Zt %comment out if you want to print @MickeyDang
```
Comment (MickeyDang) [inline testconvolute.m]: These comments will be submitted so remove them. Just state that the output is printed and can be manually compared to expected value.

File: testconvolute.m [hunk 5]
```diff
@@ -1,6 +1,43 @@
...
+Z2 %comment out if you want to print @MickeyDang
+cCheck2 %comment out if you want to print @MickeyDang
+T2 = false; %Bool for Test two
+if is_appEq(Z2,cCheck2)
+    fprintf("Test2 has passed. Convolution matches with conv(xt,ht)\n");
+    T2 = true;
+end
+%Load Summary
+fprintf("Summary of Test Results:\n")
+if T1
```
Comment (MickeyDang) [inline testconvolute.m]: Break this into a function for testing to remove repetition.

File: testfourierSeries.m [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+coefficients2 = fourierSeries(edgecase);
+edgecheck = invFourier(coefficients2,et);
+T2 = false;
+if is_appEq(edgecase, edgecheck);
+    fprintf("Second Test Passed.\n")
+    T2 = true;
+end
+fprintf("Printing summary of results.\n")
+fprintf("First test:")
+if T1
```
Comment (MickeyDang) [inline testfourierSeries.m]: Suggestion to make an `assert` function to avoid repeating the if statements.

File: testfourierSeries.m [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+oSize = length(xt);
+xtcheck = invFourier(coefficients, t);
+hold on
+plot (t,xtcheck+0.1)
+T1 = false;
+if is_appEq(xt,xtcheck);
+    fprintf("First Test Passed.\n")
+    T1 = true;
+end
+edgecase = [1,3,3,3,3,3,3,2,3,2,1,3,1,3,3,3,3,3,3,2,3,2,1,3];
```
Comment (MickeyDang) [inline testfourierSeries.m]: you don't need to say edge case. usually when we test we have each test case be its own function and we name the function a test case (for example "happyPathOne" , "repeatingDuplicatesPeriod", "periodOfOne" to denote the case in the function declaration etc)

File: testconvolute.m [hunk 8]
```diff
@@ -1,6 +1,43 @@
...
+else
+    fprintf("Test1: Failed\n")
+end
+if T2
+    fprintf("Test2: Success\n")
+else
+    fprintf("Test2: Failed\n")
+end
+    
+function a = is_appEq(arr1,arr2);
```
Comment (MickeyDang) [inline testconvolute.m]: nice job with the function to avoid repetition!

File: convolute.m [hunk 9]
```diff
@@ -1,9 +1,13 @@
...
-    for d = 1:(length(y))
-        Z(c+d-1,1) = Z(c+d-1,1) +  0.01*x(c)*y(d);
+Z = zeros(1,length(x) + length(y) - 1);
+%Using the sampling property
+%Convolution becomes length(x)*length(y) summations
+for indX = 1:(length(x))
+    for indY = 1:(length(y))
+        %The (-1) on the Z index is because MATLAB ARRAYS start at 1
+        %Indexing of c+d to store where the addition is stored
+        Z(1,indX+indY-1) = Z(1,indX+indY-1) +  0.01*x(indX)*y(indY);
```
Comment (MickeyDang) [inline convolute.m]: Hm. Lets talk about this offline i just want to understand it.

**Review by MickeyDang — COMMENTED** [review]
Some changes. But i like the improvement ✅ .

File: testconvolute.m [hunk 10]
```diff
@@ -2,42 +2,52 @@
...
 Zt = convolute(xt,ht);%function Z(t), convoluted function
 cCheck = conv(0.01*xt,ht);%Our check for our test
-Zt %comment out if you want to print @MickeyDang
-cCheck %comment out if you want to print @MickeyDang
-T1 = false; %Bool for Test 1
-if is_appEq(Zt,cCheck) %Easier to set a tolerance because MATLAB has negative 0's
-    fprintf("Test1 has passed. Convolution matches with conv(xt,ht)");
-    T1 = true;
-end
+Zt; %comment out if you want to print @MickeyDang
```
Comment (MickeyDang) [inline testconvolute.m:5]: Can you get rid of the @MickeyDang comment

File: testconvolute.m [hunk 11]
```diff
@@ -2,42 +2,52 @@
...
-Z2 = convolute(sint,cost);
-cCheck2 = conv(.01*sint,cost);
-Z2 %comment out if you want to print @MickeyDang
-cCheck2 %comment out if you want to print @MickeyDang
-T2 = false; %Bool for Test two
-if is_appEq(Z2,cCheck2)
-    fprintf("Test2 has passed. Convolution matches with conv(xt,ht)\n");
-    T2 = true;
-end
+T(2,1) = test_convolute(xt,ht,"2"); %Bool for Test two
```
Comment (MickeyDang) [inline testconvolute.m:15]: 💯 !

Comment (MickeyDang) [reply testconvolute.m:15]: There's no need to comment with the bool for test two though. i think its pretty obvious.

File: testconvolute.m [hunk 12]
```diff
@@ -2,42 +2,52 @@
...
+function check = test_convolute(xt,ht, test_name)
+    check = is_appEq(conv(0.01*xt,ht),convolute(xt,ht));
+    if check
+        fprintf("Test " + test_name + " Succeeded\n")
+    else
+        fprintf("Test " + test_name + " Failed\n")
+    end
+end
+
+function eq = is_appEq(arr1,arr2); %function to check if arr1 close toarr2
```
Comment (MickeyDang) [inline testconvolute.m:39]: can we rename it to `isApproxEqual()` rather than having a comment to an abbreviated name?

File: testconvolute.m [hunk 13]
```diff
@@ -2,42 +2,52 @@
...
+eq = true;
+if length(arr1) ~= length(arr2);
+    eq = false;
+end
+for i=1:length(arr1)
+    if (eq== false)
+        break
+    end
+    %Below is a long if statement that checks whether the diff.
```
Comment (MickeyDang) [inline testconvolute.m:48]: You don't need to write the set up "below is..." . Just say "checks whether diff is < tolerance error" or something along those lines. nice and concise.

**Review by MickeyDang — COMMENTED** [review]
Great work! Much nicer and concise code!
A few comment and style revisions requested and then we're good.
